smac/intensification/successive_halving.py

```diff
@@ -972,11 +959,13 @@ def _get_pending_instances_for_stage(self, run_history: RunHistory) -> int:
         else:
             curr_insts = self.inst_seed_pairs
 
-        # The minus one here accounts for the fact that len(curr_insts) is a length starting at 1
-        # and self.curr_inst_idx is a zero based index
-        # But when all configurations have been launched and are running in run history
-        # n_insts_remaining becomes -1, which is confusing. Cap to zero
-        n_insts_remaining = max(len(curr_insts) - self.curr_inst_idx - 1, 0)
+        # self.curr_inst_idx - 1 is the last proposed instance/seed pair from get_next_run
+        # But it is zero indexed, so (self.curr_inst_idx - 1) + 1 is the number of configurations
+        # that we have proposed to run in total for the running configuration via get_next_run
+        n_insts_remaining = len(curr_insts)
+        if self.running_challenger is not None:
+            n_insts_remaining = max(
+                len(curr_insts) - self.curr_inst_idx[self.running_challenger], 0)
```

Comment:
Why is this only calculated for `self.running_challenger` and not for all entries in `self.curr_inst_idx`?

Reply:
Or maybe the current challenger should be an argument to this function to make it easier to understand it's purpose?